### PR TITLE
Update ESlint rules and cleanup

### PR DESCRIPTION
**Pull request by jasonadt**
## Summary of Changes:

  * Updated `.editorconfig` to match the rest of our repos
  * Removed Prettier.
  * Updated ESLint rules
  * Updated GitHub NodeJS workflow for this repo to automatically check for ESLint errors for PRs (and block merging when there are errors)
  * Fixed errors raised by the new rules
  * Removed all `package.json` files except for the root one (going forward: we will only have the root one)
  * Renamed `index.js` files to match the component name (going forward: we will only have a single `index.js` file).
  * Minor cleanup
  * Renamed some components
  * Consolidated all components under a new `components` directory.

**Testing**:  
  * **Unit-Testing**: `npm run test`: **PASS**
  * **ESLint**: `npm run lint`: **PASS**
  * **Local Testing**: I've tested the docs app with `npm run start` and each of the docs is working fine with no console errors. As previously discussed, we will perform a thorough integration test with the `integration/ui-improvements` branch after this PR has been merged (as part of merging that branch to `master`)

## Checks
- [X] Requires documentation update
- [X] Requires lib version update

## Versioning impact
- [X] **MAJOR** incompatible API changes.
- [ ] **MINOR** adds functionality in a backwards-compatible manner.
- [ ] **PATCH** backwards-compatible bug fixes.

### CODE REVIEW

File: src/components/Button/Button.jsx [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  const [isPressed, setIsPressed] = useState(false);
+
+  const onHandleClick = (event) => {
+    if (onClick) { onClick(event); }
+    if (isPressed) { setIsPressed(!isPressed); }
+    event.preventDefault();
+  };
+
+  const changePressedStatus = () => setIsPressed(!isPressed);
+  const typeClass = validTypes.includes(type) ? CSS[type] : '';
```
Comment (jantocv) [inline src/components/Button/Button.jsx:19]:
```suggestion
  const typeClass = (validTypes.includes(type)) ? CSS[type] : '';
```

File: src/components/ListItem/ListItem.jsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import PropTypes from 'prop-types';
+
+import CSS from './styles.css';
+import Heading from '../Heading/Heading';
+import TextParagraph from '../TextParagraph/TextParagraph';
+import ToggleSwitch from '../ToggleSwitch/ToggleSwitch';
+
+const ListItem = ({ title, icon, iconColor, description, className, onLink, linkText, onChange, checked, dataTestId }) => (
+  <li className={`${CSS.item} ${className}`} data-testid={dataTestId}>
+    {(icon) ? (<div data-testid={`${dataTestId}-icon`} className={`${CSS.media} ${iconColor ? CSS[iconColor] : ''}`}>{icon}</div>) : null}
```
Comment (jantocv) [inline src/components/ListItem/ListItem.jsx:10]:
```suggestion
    {(icon) ? (<div data-testid={`${dataTestId}-icon`} className={`${CSS.media} ${(iconColor) ? CSS[iconColor] : ''}`}>{icon}</div>) : null}
```

File: src/components/PhoneNumberInputField/PhoneNumberInputField.jsx [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    setPhoneValue(formattedValue);
+    phoneField.value = formattedValue;
+    if (onBlur) { onBlur(formattedValue); }
+  };
+
+  const getPrefix = () => `+${getCountryCallingCode(countryCode, Metadata)}`;
+
+  const result = useMemo(() => {
+    const prefix = getPrefix();
+    const filledClass = showCountryCode ? CSS.filled : '';
```
Comment (jantocv) [inline src/components/PhoneNumberInputField/PhoneNumberInputField.jsx:43]:
```suggestion
    const filledClass = (showCountryCode) ? CSS.filled : '';
```

File: src/components/RadioList/RadioList.jsx [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+  const [currentValue, setCurrentValue] = useState(null);
+  const callbackDelayInMilliseconds = 250;
+
+  useEffect(() => {
+    if (value) {
+      setCurrentValue(value);
+      return;
+    }
+    const defaultCheckedOptionIndex = options.findIndex(({ selected }) => selected);
+    setCurrentValue(defaultCheckedOptionIndex !== -1 ? options[defaultCheckedOptionIndex].value : null);
```
Comment (jantocv) [inline src/components/RadioList/RadioList.jsx:17]:
```suggestion
    setCurrentValue((defaultCheckedOptionIndex !== -1) ? options[defaultCheckedOptionIndex].value : null);
```

File: src/components/RadioList/RadioList.jsx [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+      <ul data-testid={`${dataTestId}-select`}>
+        {options.map(({ label, value: optionValue, children, onClick }, index) => {
+          const elementId = `${id}-${index}`;
+          const checked = (currentValue === optionValue);
+          return (
+            <li
+              data-testid={`${dataTestId}-option`}
+              key={index}
+              data-checked={checked}
+              className={`${CSS.item} ${checked ? CSS.checked : ''}`}
```
Comment (jantocv) [inline src/components/RadioList/RadioList.jsx:40]:
```suggestion
              className={`${CSS.item} ${(checked) ? CSS.checked : ''}`}
```

File: src/components/Button/Button.jsx (hunk 1 above)
Comment (jasonadt) [reply src/components/Button/Button.jsx:19]: Thanks @jantocv, good catch! :)

**Pull request merged**